storage_service/locations/tests/test_package.py

```diff
@@ -357,6 +357,27 @@ def test_run_post_store_callbacks_dip(self):
             body = '{"download_url": "http://ss.com/api/v2/file/%s/download/"}' % uuid
             mocked_execute.assert_called_with(url, body)
 
+    def test_replicate_aip(self):
+        space_dir = tempfile.mkdtemp(dir=self.tmp_dir, prefix="space")
+        replication_dir = tempfile.mkdtemp(dir=self.tmp_dir, prefix="replication")
+        aip = models.Package.objects.get(uuid="0d4e739b-bf60-4b87-bc20-67a379b28cea")
+        aip.current_location.space.staging_path = space_dir
+        aip.current_location.space.save()
+        aip.current_location.replicators.create(
+            space=aip.current_location.space,
+            relative_path=replication_dir,
+            purpose=models.Location.REPLICATOR,
+        )
+
+        aip.create_replicas()
+        replica = aip.replicated_package
+
+        assert replica is not None
+        # The relationship is a little bit broken here; one would think that the
+        # _original_ AIP should have replicas, and the replicated AIP should not.
```

Comment:
I agree! The database is a little cleaner in this regard, and just has a `replicated_package_id` so I guess we just work it out from that.

Reply:
It looks like `replicated_package_id` is also backwards (I think unintentionally, based on the [docstring of replicate](https://github.com/artefactual/archivematica-storage-service/blob/stable/0.14.x/storage_service/locations/models/package.py#L468)), that is, the original package will have a value for `replicated_package_id`, and the replica package will not. This seems to me to be the cause of https://github.com/artefactual/archivematica-storage-service/issues/270. It needs to be looked at more though.